llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1901,6 +1907,13 @@ def cmp_combines: GICombineGroup<[
   redundant_binop_in_equality
 ]>;
 
+
+def artifact_combines: GICombineGroup<[
+  merge_combines,
+  match_addos,
```

Comment:
These aren't artifacts.

Reply:
Well, the legalizer builds addos while narrowing, e.g., adds:
https://github.com/llvm/llvm-project/blob/764275949897533a4be0728250e69a94d228fbc5/llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp#L6289

Maybe artifacts is too strict. I stated in the summary that we import addos and the legalizer builds while legalizing non addos.

Reply:
That doesn't make them artifacts. There are lots of instructions that the legalized builds like loads/stores but those aren't artifacts either.

Reply:
Ok, I need a mechanism to combine addos in AArch64PostLegalizerCombiner and I want to have a group where I/we can add more combines that run in the post legalizer  combiner. It is sufficient to rename `artifact_combines` to something like `post_legalizer_combines` ?

Reply:
Just create a new group for these? And then add the group to the post-legalizer combiner set.

Reply:
Shouldn't name these as post-legalizer combines in this file. Each target defines their own pipeline. Name the groups based on their category and then refer to them in any custom groups needed within AArch64Combine.td

Reply:
I splitted them again into two topic groups.